ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -5,13 +5,57 @@
 package com.mozilla.telemetry.transforms;
 
 import com.google.api.services.bigquery.model.TableRow;
+import com.google.common.collect.Maps;
 import com.mozilla.telemetry.util.Json;
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.io.gcp.bigquery.TableDestination;
+import org.apache.beam.sdk.io.gcp.bigquery.TableDestinationCoderV2;
+import org.apache.beam.sdk.io.gcp.bigquery.TableRowJsonCoder;
 import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.commons.text.StringSubstitutor;
 
-public class PubsubMessageToTableRow extends MapElementsWithErrors<PubsubMessage, TableRow> {
+public class PubsubMessageToTableRow
```

Comment:
The name of this class is now a bit of a stretch, since it returns a KV, but I couldn't think of any clearer alternative.